### DISCUSSION TITLE
Selecting json keys with special characters (escaping?)

### DISCUSSION THREAD

**Discussion by coffepowered** [duckdb/duckdb#7326] [General]
Hello,
It seems that selecting json fields with special characters is not currently supported.
This code:
```
SELECT key, json_column->>key FROM table
```

would work perfectly on the record `test_key, {"test_key": 1}`, but fail when test_key contains special characters, e.g. `test/key, {"test/key": 1}`.

Do you have any suggestions? I tried escaping the query with backticks, but that does not seem to work.

**Comment by coffepowered** [comment]
for anyone reading, the problem may be solved by escaping the key in the following fashion:
> json_extract(json_column, '$."key"') as extracted_content

If "key" needs to be referenced dynamically from another field, that can be modified like this:
> json_extract(json_column, CONCAT('$."', "dynamic_key_reference",'"') as extracted_content

This would probably deserve a mention in the documentation or could be an inspiration for new functionalities.

**Reply by lnkuiper** [reply]
Hi @coffepowered , the `/` character is special here because we support two extract syntaxes.

The first one is the JSON pointer notation, where `/` is a special character:
```sql
select '{"test_key": 1}'->>'/test_key' k;
┌─────────┐
│    k    │
│ varchar │
├─────────┤
│ 1       │
└─────────┘
```

The other one is the JavaScript-style extract, where `$`, `.`, `[` and `]` are special characters:
```sql
select '{"test/key": 1}'->>'$.test/key';
┌─────────┐
│    k    │
│ varchar │
├─────────┤
│ 1       │
└─────────┘
```

This is described [in our documentation](https://duckdb.org/docs/extensions/json.html#json-scalar-functions), but perhaps too brief.

**Reply by coffepowered** [reply]
Oh, I see. In fact, I did not realize the special characters for each extraction method.
Thanks for your support. My application seems to be quite unfortunate, as I have to support both `.` and `/`. Reworked example :) :

`select '{"test/.key": 1}'->>'$."test/.key"';`

**Reply by lnkuiper** [reply]
This works!
```sql
select '{"test/.key": 1}'->>'$."test/.key"' k;
┌─────────┐
│    k    │
│ varchar │
├─────────┤
│ 1       │
└─────────┘
```
Double quotes `""` can be used to escape characters in the `$` extraction mode. This *__definitely__* should be documented.

Thanks for the discussion. I will add this to the documentation before the next DuckDB release! :)
